### PR TITLE
Support setting random seed in evaluation key

### LINKED ISSUE #452: [Feature request]:  gordo exposes fixing the seed to the user
At the moment gordo does not support fixing the seed in order to reproduce results.

Example:
This code does not fix the seed. 

`numpy.random.seed(42)`
`tensorflow.set_random_seed(42)`

If you re train a gordo model, it will produce different results.

The ability to reproduce results is important to study the influence of the hyperparameters on a model.

### PR DESCRIPTION
**Pull request by milesgranger**
Problem :thinking: 
---
_Sometimes_ it is appropriate to set the seed when training/evaluating models.

Solution :open_hands: 
---
Add ability to do the following in a config in the `evaluation` key:
```yaml
evalutation:
  seed: 1234
```

Will close #452

### CODE REVIEW

File: gordo_components/builder/build_model.py [hunk 1]
```diff
@@ -88,6 +90,13 @@ def build_model(
     -------
         Tuple[Optional[sklearn.base.BaseEstimator], dict]
     """
+    if evaluation_config.get("seed"):
+        seed = evaluation_config["seed"]
+        logger.info(f"Setting random seed: '{seed}'")
+        tf.random.set_seed(seed)
+        np.random.seed(seed)
+        random.seed(seed)
```
Comment (ryanjdillon) [inline gordo_components/builder/build_model.py:98]: Would it be worth testing that these all infact got set to `seed`?  Maybe if these libs change how things get set, and these calls don't act as planned in a way that wouldn't affect the `df1==df2` test but could affect some other pipeline not represented in the tests.

**Comment by ryanjdillon** [comment]
I don't recall what was landed on, or perhaps not present with the final decision of things, with regards to how exposed the seed setting should be. Otherwise, this LGTM! :icecream:

**Review by ryanjdillon — APPROVED** [review]

File: gordo_components/builder/build_model.py [hunk 2]
```diff
@@ -88,6 +90,13 @@ def build_model(
     -------
         Tuple[Optional[sklearn.base.BaseEstimator], dict]
     """
+    if evaluation_config.get("seed"):
```
[the remark this answers is not in the document]
Comment (milesgranger) [reply gordo_components/builder/build_model.py:93]: Same for me :man_shrugging:

File: gordo_components/builder/build_model.py (hunk 1 above)
Comment (milesgranger) [reply gordo_components/builder/build_model.py:98]: It would be _very_ strange indeed if the api changed in such a way the seed was _not_ set and yet the test as it is, still passed. 

anyhow, according to _some guy_, it doesn't seem (that) possible. 
> [The short answer is that you simply can't (at least not in general).](https://stackoverflow.com/questions/32172054/how-can-i-retrieve-the-current-seed-of-numpys-random-number-generator)

But manually looking is appears one could (for numpy anyway) index into the state `np.random.get_state()[1][0]` but I don't think it's worth it.

**Comment by EinarJohnsen** [comment]
Perhaps a bit late, but should this also be tested with a NN Keras model? I've heard there is problems with neural networks earlier on, not sure if it is fixed after TF 2.0. @milesgranger

**Comment by milesgranger** [comment]
Good point, I added it and seems a-ok. So TF 2.0 ftw :checkered_flag:

File: gordo_components/builder/build_model.py (hunk 1 above)
Comment (ryanjdillon) [reply gordo_components/builder/build_model.py:98]: Right on :+1:  Sounds to good me. The comments in *some dude*'s answer got *heated*.
